protoc-gen-openapiv2/internal/genopenapi/template_test.go

```diff
@@ -1902,11 +1902,10 @@ func TestApplyTemplateRequestWithBodyQueryParameters(t *testing.T) {
 				Number: proto.Int32(1),
 			},
 			{
-				Name:     proto.String("book"),
-				Label:    descriptorpb.FieldDescriptorProto_LABEL_REQUIRED.Enum(),
-				Type:     descriptorpb.FieldDescriptorProto_TYPE_STRING.Enum(),
-				TypeName: proto.String("Book"),
-				Number:   proto.Int32(2),
+				Name:   proto.String("book"),
+				Label:  descriptorpb.FieldDescriptorProto_LABEL_REQUIRED.Enum(),
+				Type:   descriptorpb.FieldDescriptorProto_TYPE_STRING.Enum(),
```

Comment:
@johanbrandhorst I have a question about this line. The original implementation uses `FieldDescriptorProto_TYPE_STRING` and I kept this unchanged, however, it also provides `TypeName` equal to `Book`. The new protogen-based implementation fails with this configuration which makes sense because if we want `TypeName` equal to `Book` it clearly is a message type.

I am not sure what was the original intention because either `TypeName` is not needed or it is needed but then the `Type` should be `FieldDescriptorProto_TYPE_MESSAGE`. If the latter is the original intention, the code fails in the same way in this PR and on the master branch, hence, I assumed the former is the right one.

Reply:
That sounds right to me, frankly I haven't looked at this part of the code for a while, so I trust you have it right.